### PR TITLE
implement BIT b,r

### PR DESCRIPTION
**Pull request by ginglis13**
Issue #20 

pretty similar logic to `SET b,r`. Just `AND` some hex to check if a bit is zero - if so, set the `ZERO` flag.

### CODE REVIEW

**Comment by ginglis13** [comment]
@sambattalio do you want me to fix this as per #23 ?

**Comment by sambattalio** [comment]
@ginglis13 yes please. Also, I added a new macro SET_REGISTER which might help out. It's all over if you need an example/

**Comment by ginglis13** [comment]
I'm not sure i'll use the `SET_REGISTER` macro since all that's need is to `SET_ZERO` if the check is just that the bit `b` in reg `r` is 0.

I do have a question about the HL register - I'm leaving it as a reference in `p->memory` since it looks like as you said in that issue that `(HL)` is dereferencing memory. 

Pushing those changes, lmk yo

File: src/proc.c [hunk 1]
```diff
@@ -2844,6 +3268,10 @@ void proc_handle_cb_prefix(Proc *p) {
             RESET_SUBTRACT;
             SET_HALF_CARRY;
             debug_print("BIT 0,H\n", NULL);
+
+            if( p->memory[p->registers.h] == (p->memory[p->registers.h] & 0xFE) ){
```
Comment (sambattalio) [inline src/proc.c]: Yo I think there are a few more you might need to grab if you don't mind

File: src/proc.c [hunk 2]
```diff
@@ -2836,6 +3256,10 @@ void proc_handle_cb_prefix(Proc *p) {
             RESET_SUBTRACT;
             SET_HALF_CARRY;
             debug_print("BIT 0,E\n", NULL);
+
+            if( p->memory[p->registers.e] == (p->memory[p->registers.e] & 0xFE) ){
```
Comment (sambattalio) [inline src/proc.c]: another one

**Review by sambattalio — CHANGES_REQUESTED** [review]
Yo I think theres a few more if you dont mind grabbing them. Like ~CB 0x40ish

File: src/proc.c (hunk 2 above)
Comment (ginglis13) [reply src/proc.c]: good call yo

**Comment by sambattalio** [comment]
I’ll peep this in the morning ty ginglis

**Comment by sambattalio** [comment]
think you got it nice thanks

**Comment by ginglis13** [comment]
@sambattalio i think later today I'm gonna add tests for this and maybe some other instructions

**Comment by sambattalio** [comment]
oh that sounds good. But you might wanna wait a second to do that. I was thinking of doing test ROMs for the instruction testing instead of testing in C. thought that would be more fun. Im gonna try to get it set up today and I'll lyk it shouldn't be bad (probably should be easier than C)
@ginglis13

**Comment by ginglis13** [comment]
awesome sounds dope @sambattalio

**Pull request merged**